src/server/page_server.cpp

```diff
@@ -101,23 +101,26 @@ void page_server::receive_log_page_fetch (cubpacking::unpacker &upk)
       _er_log_debug (ARG_FILE_LINE, "Received request for log from Transaction Server. Page ID: %lld \n", pageid);
     }
 
-  assert (m_log_page_fetcher);
-  m_log_page_fetcher->fetch_page (pageid, std::bind (&page_server::on_log_page_read_result, this, std::placeholders::_1,
+  assert (m_page_fetcher);
+  m_page_fetcher->fetch_log_page (pageid, std::bind (&page_server::on_log_page_read_result, this, std::placeholders::_1,
 				  std::placeholders::_2));
 }
 
 void page_server::receive_data_page_fetch (cubpacking::unpacker &upk)
 {
-  if (prm_get_bool_value (PRM_ID_ER_LOG_READ_DATA_PAGE))
-    {
-      VPID vpid;
-      std::string message;
+  std::string message;
+  upk.unpack_string (message);
 
-      upk.unpack_string (message);
-      std::memcpy (&vpid, message.c_str (), sizeof (vpid));
-      _er_log_debug (ARG_FILE_LINE, "Received request for Data Page from Transaction Server. pageid: %ld volid: %d\n",
-		     vpid.pageid, vpid.volid);
-    }
+  VPID vpid;
+  std::memcpy (&vpid, message.c_str (), sizeof (vpid));
+
+  LOG_LSA nxio_lsa;
```

Comment:
Better to name according to the purpose of the LSA, rather than what its value is set to now. The value is subject of change, its purpose is not.

```suggestion
  LOG_LSA target_repl_lsa;
```